router/core/subscriptions_modules.go

```diff
@@ -91,19 +91,31 @@ func (c *pubSubSubscriptionOnStartHookContext) WriteEvent(event datasource.Strea
 	return true
 }
 
+type UnsafeEngineEvent []byte
+
+func (e UnsafeEngineEvent) GetData() []byte {
+	return e
+}
+
+func (e UnsafeEngineEvent) SetData(data []byte) {
+	copy(e, data)
+}
```

Comment:
This will cause a bug because while copy swaps out the underlying byte data on the existing memory location, it only copies whatever is shorter (`len(e)` vs `len(data)`): https://go.dev/play/p/umbTn0a-NFa

I think the only safe way to do it is pointer receivers.